tree/ntuple/v7/test/ntuple_ioerrors.cxx

```diff
@@ -0,0 +1,403 @@
+   char buffer[bufferSize];
+
+   context.fFailureType = RRawFile::EFailureType::kShortRead;
+   context.fRangeBegin = 0;
+   context.fRangeEnd = s.length();
+   
+   RRawFile::ROptions options;   
+   auto rrawfileinstance = RRawFile::Create(fileGuard.GetPath(), options);
+   
+   size_t bytesRead = rrawfileinstance->ReadAt(&buffer, bufferSize, 0);
+
+   EXPECT_FALSE(bytesRead == bufferSize && memcmp(buffer, s.c_str(), bufferSize) == 0);
+}
+
+TEST(RRawFile, TriggerBitFlip)
+{
+   auto &context = RRawFile::GetFailureInjectionContext();
+   constexpr char const* kNTupleFileName = "bitFlipTest.root";
+   FileRaii fileGuard(kNTupleFileName);
+
+   std::ofstream ostrm(fileGuard.GetPath(), std::ios::binary);
+   std::string s = "Hello, this is a test message.";
+   ostrm.write(s.c_str(), s.length());
+   ostrm.close();
+
+   size_t bufferSize = s.length();
+   char buffer[bufferSize];
+
+   context.fFailureType = RRawFile::EFailureType::kBitFlip;
+   context.fRangeBegin = 0;
+   context.fRangeEnd = s.length();
+   context.fOccurrenceProbability = 1;
+   
+   RRawFile::ROptions options;   
+   auto rrawfileinstance = RRawFile::Create(fileGuard.GetPath(), options);
+   
+   size_t bytesRead = rrawfileinstance->ReadAt(&buffer, bufferSize, 0);
+
+   EXPECT_FALSE(bytesRead == bufferSize && memcmp(buffer, s.c_str(), bufferSize) == 0);
+}
+/* 
+   END RRAWFILE TESTS
+ */
+
+/* 
+   RNTUPLE BIT FLIP TESTS
+ */
+TEST(RNTuple, UncompressedBitFlipHeader)
+{
+   auto &context = RRawFile::GetFailureInjectionContext();
+   constexpr char const* kNTupleFileName = "test_ntuple.root";
+   constexpr char const* kNTupleName = "ntuple";
+   FileRaii fileGuard(kNTupleFileName);
+   int compression = 0;
+
+   // set up the ntuple   
+   {
+      auto model = RNTupleModel::Create();
+      auto ptrPt = model->MakeField<float>("pt");
+      auto ptrE = model->MakeField<float>("energy");
+
+      RNTupleWriteOptions options;
+      options.SetCompression(compression);
+      auto writer = RNTupleWriter::Recreate(std::move(model), kNTupleName, fileGuard.GetPath(), options);
+
+      for (size_t i = 0; i < 10000; ++i){
+         *ptrPt = i;
+         *ptrE = i;
+         writer->Fill(); 
+      }
+   }
+
+   // set up the tester
+   auto f = TFile::Open(fileGuard.GetPath().c_str(),"READ");
+   auto ntpl = f->Get<ROOT::Experimental::RNTuple>(kNTupleName);
+   RNTupleTester tester(*ntpl);
+
+   // set failure parameters
+   context.fFailureType = RRawFile::EFailureType::kBitFlip;
+   context.fRangeBegin = tester.GetSeekHeader(tester);
+   context.fRangeEnd = tester.GetSeekHeader(tester) + tester.GetNBytesHeader(tester);
+   context.fOccurrenceProbability = 1;
+
+   // attempt to open the file  
+   try {      
+      auto reader = RNTupleReader::Open(kNTupleName, fileGuard.GetPath());
+    
+      FAIL() << "bit flips in header should throw";
+   } catch (const RException& err) {
+      EXPECT_THAT(err.what(), testing::HasSubstr("CRC32 checksum mismatch"));
```

Comment:
I'm wondering whether this always throws (even if the bit is flipped somewhere in the `TKey` header)?